packages/app-builder/src/routes/_builder+/cases+/inboxes.$inboxId.tsx

```diff
@@ -70,74 +96,30 @@ export async function loader({ request, params }: LoaderFunctionArgs) {
 }
 
 export default function Cases() {
-  const { casesData: initialCasesData, filters: initialFilters } =
+  const { t } = useTranslation(casesI18n);
+  const { casesData: initialCasesData, filters } =
     useLoaderData<typeof loader>();
   const inboxId = useParam('inboxId');
-  const { t } = useTranslation(casesI18n);
 
-  const [casesData, setCasesData] = useState(initialCasesData);
-  const { items: cases, ...pagination } = casesData;
+  const { data, next, previous, reset, update } = useCursorPaginatedFetcher<
+    typeof loader,
+    PaginatedResponse<Case>
+  >({
+    transform: (fetcherData) => fetcherData.casesData,
+    initialData: initialCasesData,
+    getQueryParams: (cursor) => buildQueryParams(filters, cursor),
+    validateData: (data) => data.items.length > 0,
+  });
+  const { items: cases, ...pagination } = data;
 
-  const [filters, setFilters] = useState(initialFilters);
-  if (initialFilters !== filters) {
-    setFilters(initialFilters);
-    setCasesData(initialCasesData);
+  const [previousInitialCasesData, setPreviousInitialCasesData] =
+    useState(initialCasesData);
+  if (initialCasesData !== previousInitialCasesData) {
+    setPreviousInitialCasesData(initialCasesData);
+    update(initialCasesData);
```

Comment:
IIUC, this is to sync the data comming from `fetcher` with the internal state (updated by the internal `useFetcher`).

You get the "reset pagination state" covered when filters changed thanks to the `reset()` call above inside `navigateCasesList`.

Test this "side effect" :
1. Open the app, and do some pagination
2. Loose the focus (ex: go to another app or tab)
3. Go back to the app tab (the page loader should have been retriggered)
4. Try to play with pagination (the old state may be persisted here)

My questions :
1. Do we need to reset the state in that case too ? (or am I missing something ?)
2. This looks something we could wrap inside the `useCursorPaginatedFetcher` hook (since the sync looks pretty common to any usage)

Reply:
For the side effect:
I didn't know remix was re-running loaders when switching tabs, I thought it was only done when refreshing the page or navigating the app.
I'll fix this issue

For your questions:
1. Reseting the state is in my sense logical as you modify the filters you do not want to stay on the same pagination as you have a high-chance of modifying the dataset displayed. (The id you've on cursor with current filter may not be available for another filters)
2. Yeah that's what I did initially, but as you can see the URL based navigation and data are managed outside of the scope of the hook and I wanted to keep it that way too for the data update.